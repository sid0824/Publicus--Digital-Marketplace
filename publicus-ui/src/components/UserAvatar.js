import React from "react";
import {withAuthContext} from "../AuthContext";

const defaultAvatar = 'http://s3.amazonaws.com/37assets/svn/765-default-avatar.png';

// Component used for rendering the avatar for a user
export class UserAvatar extends React.Component {

  state = {
    url: ''
  };

  // Use the supplied props url or default to the current users avatar
  componentDidMount = async () => {
    const { image, avatar } = this.props;
    this.setState({
      url: image || avatar
    })
  };

  // Default avatar on error
  onError = () => {
    this.setState({
      url: defaultAvatar
    })
  };

  render() {
    const { url } = this.state;
    const { height, width } = this.props;
    return (
      <img
        id={"avatar"}
        alt={"avatar"}
        width={height || 32}
        height={width || 32}
        style={{ borderRadius: '30%' }}
        src={url || defaultAvatar}
        onError={this.onError}
      />
    )
  }
}

export default withAuthContext(UserAvatar);