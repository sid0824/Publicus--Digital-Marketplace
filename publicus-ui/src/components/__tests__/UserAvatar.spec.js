import toJson from 'enzyme-to-json';
import React from "react";
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { UserAvatar } from "../UserAvatar";

configure({ adapter: new Adapter() });

describe('User Avatar', () => {

  beforeEach(() => {

  });

  afterAll(() => {

  });

  test('should render current user avatar', () => {
    const wrapper = shallow(
      <UserAvatar
        height={100}
        width={100}
        avatar={"https://mock-path.png"}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });


  test('should default form url on error', () => {
    const wrapper = shallow(
      <UserAvatar
        height={100}
        width={100}
        avatar={"https://mock-path.png"}
      />
    );
    wrapper.instance().onError();
    expect(wrapper.state()).toMatchSnapshot();
  });


});