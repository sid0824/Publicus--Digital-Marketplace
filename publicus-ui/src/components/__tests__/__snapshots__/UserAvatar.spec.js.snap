// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Avatar should default form url on error 1`] = `
Object {
  "url": "http://s3.amazonaws.com/37assets/svn/765-default-avatar.png",
}
`;

exports[`User Avatar should render current user avatar 1`] = `
<img
  alt="avatar"
  height={100}
  id="avatar"
  onError={[Function]}
  src="https://mock-path.png"
  style={
    Object {
      "borderRadius": "30%",
    }
  }
  width={100}
/>
`;
